nixos/modules/services/cluster/k3s/default.nix

```diff
@@ -114,6 +115,26 @@ in
       default = null;
       description = "File path containing the k3s YAML config. This is useful when the config is generated (for example on boot).";
     };
+
+    registries = mkOption {
+      type = types.nullOr yaml.type;
+      default = null;
+      example = {
+        mirrors."docker.io".endpoint = ["https://registry.example.com:5000"];
+        configs."registry.example.com:5000" = {
+          auth = {
+            username = "";
+            password = "";
```

Comment:
Storing secrets in the nix store is bad practice in general. At least, we should be able to provide an alternative such as `passwordFile`, so users can drop a file somewhere and we should be able to use it.

Reply:
D'oh.

I'll need time to come up with a solution for this. It will likely involve a layer of indirection—e.g. [a shell script](https://github.com/NixOS/nixpkgs/blob/27c13997bf450a01219899f5a83bd6ffbfc70d3c/nixos/modules/services/matrix/mjolnir.nix#L60-L62)—and decomposition of this option into its constituent parts, `configs`, `mirrors`, `configs.*.auth`, `configs.*.auth.passwordFile`, and so on.